tsl/src/dist_util.c

```diff
@@ -183,65 +186,34 @@ Datum
 dist_util_remote_hypertable_info(PG_FUNCTION_ARGS)
 {
 	char *node_name;
-	FuncCallContext *funcctx;
-	PGresult *result;
-
-	Assert(!PG_ARGISNULL(0)); /* Strict function */
+	StringInfo query_str = makeStringInfo();
+	/* Strict function */
+	Assert(!PG_ARGISNULL(0) && !PG_ARGISNULL(1) && !PG_ARGISNULL(2));
+	Name schema_name = PG_GETARG_NAME(1);
+	Name table_name = PG_GETARG_NAME(2);
+	appendStringInfo(query_str,
+					 "SELECT * from _timescaledb_internal.hypertable_local_size( '%s', '%s'  );",
+					 quote_identifier(NameStr(*schema_name)),
+					 quote_identifier(NameStr(*table_name)));
```

Comment:
Function `quote_identifier` cannot be used here. If the identifier contain a single quote (`'`) you will get the strange output `'"foo'bar"'`. Use `quote_literal_cstr` instead.
```suggestion
	appendStringInfo(query_str,
					 "SELECT * from _timescaledb_internal.hypertable_local_size(%s, %s);",
					 quote_literal_cstr(NameStr(*schema_name)),
					 quote_literal_cstr(NameStr(*table_name)));
```

Reply:
Good catch. fixed and added a test.